ERCS/erc-7786.md

```diff
@@ -0,0 +1,217 @@
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Message Field Encoding
+
+A cross-chain message consists of a sender, receiver, payload, and list of attributes.
+
+#### Sender & Receiver
+
+The sender account (in the source chain) and receiver account (in the destination chain) MUST be represented using CAIP-10 account identifiers. Note that these are ASCII-encoded strings.
+
+A CAIP-10 account identifier embeds a CAIP-2 chain identifier along with an address. In some parts of the interface, the address and the chain parts will be provided separately rather than as a single string, or the chain part will be implicit.
+
+#### Payload
+
+The payload is an opaque `bytes` value.
+
+#### Attributes
+
+Attributes are structured pieces of message data and/or metadata. Each attribute is a key-value pair, where the key determines the type and encoding of the value, as well as its meaning and behavior.
+
+Some attributes are message data that must be sent to the receiver, although they can be transformed as long as their meaning is preserved. Other attributes are metadata that will be used by the intervening gateways and potentially removed before the message reaches the receiver.
+
+The set of attributes is extensible. It is RECOMMENDED to standardize attributes and their characteristics by publishing them as ERCs.
+
+A gateway MAY support any set of attributes. An empty attribute list MUST always be accepted by a gateway.
+
+Each attribute key MUST have the format of a Solidity function signature, i.e., a name followed by a list of types in parentheses. For example, `minGasLimit(uint256)`.
+
+In this specification attributes are encoded as an array of `bytes` (i.e., `bytes[]`). Each element of the array MUST encode an attribute in the form of a Solidity function call, i.e., the first 4 bytes of the hash of the key followed by the ABI-encoded value.
+
+##### Standard Attributes
+
+The following standard attributes MAY be supported by a gateway.
+
+- `postProcessingOwner(address)`: The address of the account that shall be in charge of message post-processing.
+
+### Source Gateway
+
+An Source Gateway is a contract that offers a protocol to send a message to a receiver on another chain. It MUST implement `IERC7786GatewaySource`.
+
+```solidity
+interface IERC7786GatewaySource {
+    event MessageCreated(bytes32 indexed outboxId, string sender, string receiver, bytes payload, uint256 value, bytes[] attributes);
+    event MessageSent(bytes32 indexed outboxId);
+
+    error UnsupportedAttribute(bytes4 signature);
+
+    function supportsAttribute(bytes4 signature) external view returns (bool);
+
+    function sendMessage(
+        string calldata destination, // CAIP-2 chain identifier
+        string calldata receiver, // CAIP-10 account address
```

Comment:
I think it would be clearer to call this destinationChain and destinationAddress. Without the comments, it's unclear to me which is the chain versus address.

Reply:
Thats is documented on line 35 of this document, and it matches the wording in this document https://github.com/0xFableOrg/xchain/tree/7789933bba24e2dd893cf515157af70474e7180b

That being says, I think your changes are not bad. Its a balance between clarity and conciseness

Reply:
What would you think of
- `sourceChain`
- `sender`
- `destinationChain`
- `receiver`
?

Reply:
I think that is clearer, at least the chain fields are explicit, and therefore the other fields meaning can be inferred

Reply:
UQAxBkv-DqOjrhJ7s5J6C9w3WqbSwB9H-kJc0cI-ElSbzi0l